owner-tool/src/main.rs

```diff
@@ -797,55 +797,69 @@ async fn report_to_rendezvous(matches: &ArgMatches<'_>) -> Result<(), Error> {
     let mut owner_addresses = owner_addresses.context("Error parsing owner addresses")?;
-    let owner_addresses = owner_addresses.drain(..).flatten().collect();
+    let owner_addresses: Vec<TO2AddressEntry> = owner_addresses.drain(..).flatten().collect();
 
     // Determine the RV IP
-    let rv_info = ov
-        .get_header()
-        .context("Error getting OV header")?
+    let rv_info = ov_header
         .rendezvous_info
         .to_interpreted(RendezvousInterpreterSide::Owner)
         .context("Error parsing rendezvous directives")?;
     if rv_info.is_empty() {
         bail!("No rendezvous information found that's usable for the owner");
     }
-    // Use the first entry
-    let rv_info = rv_info.first().unwrap();
-    let rv_urls = rv_info.get_urls();
-    if rv_urls.is_empty() {
-        bail!("No usable rendezvous URLs were found");
-    }
-    let rv_url = rv_urls.first().unwrap();
-
-    println!("Using rendezvous server at url {}", rv_url);
+    let mut rendezvous_performed = false;
+    for rv_directive in rv_info {
+        let rv_urls = rv_directive.get_urls();
+        if rv_urls.is_empty() {
+            bail!("No usable rendezvous URLs were found");
+        }
 
-    let mut rv_client = fdo_http_wrapper::client::ServiceClient::new(rv_url);
+        for rv_url in rv_urls {
+            println!("Using rendezvous server at url {}", rv_url);
 
-    // Send: Hello, Receive: HelloAck
-    let hello_ack: RequestResult<messages::to0::HelloAck> = rv_client
-        .send_request(messages::to0::Hello::new(), None)
-        .await;
-    let hello_ack = hello_ack.context("Error requesting nonce from rendezvous server")?;
+            let mut rv_client = fdo_http_wrapper::client::ServiceClient::new(&rv_url);
 
-    // Build to0d and to1d
-    let to0d = TO0Data::new(ov, wait_time, hello_ack.nonce3().clone());
-    let to0d_vec = serde_cbor::to_vec(&to0d).context("Error serializing to0d")?;
-    let to0d_hash = Hash::new(None, &to0d_vec).context("Error hashing to0d")?;
-    let to1d_payload = TO1DataPayload::new(owner_addresses, to0d_hash);
-    let to1d =
-        COSESign::new(&to1d_payload, None, &owner_private_key).context("Error signing to1d")?;
+            // Send: Hello, Receive: HelloAck
+            let hello_ack: RequestResult<messages::to0::HelloAck> = rv_client
+                .send_request(messages::to0::Hello::new(), None)
+                .await;
 
-    // Send: OwnerSign, Receive: AcceptOwner
-    let accept_owner: RequestResult<messages::to0::AcceptOwner> = rv_client
-        .send_request(messages::to0::OwnerSign::new(to0d, to1d), None)
-        .await;
-    let accept_owner = accept_owner.context("Error registering self to rendezvous server")?;
+            let hello_ack = match hello_ack.context("Error requesting nonce from rendezvous server")
+            {
+                Ok(hello_ack) => hello_ack,
+                Err(e) => {
+                    println!("We're inside the error: {:?}", e);
```

Comment:
Let’s use `log::info!` for this, and provide a but more information on what we were trying to accomplish?

Reply:
Oh, that was me forgetting a debugging message lol.

In order to use `log::info`, we'll need to add `log` and `pretty_env_logger` as dependencies. If it's OK, what I can do is create a different PR and migrate all the `println` in the Owner Tool to `log::{info, error, trace}`, etc.

In any case, what do you think of the following match statement?

```
let hello_ack = match hello_ack
            {
                Ok(hello_ack) => hello_ack,
                Err(e) => {
                    log::error!("Error requesting nonce from rendezvous server: {:?}", e);
                    continue;
                }
            };
```

Reply:
Yep, that works. And let's for now just import `log` and not yet `pretty_env_logger`: PR #50 centralizes the initialization of logging.
I don't think we want to replace every `println` with a `log`: the actual output of the command the user executed should stay a `println!`. (but yes, there is a reasonable amount of things that should become `log`).